### PR TITLE
Link to the Blockonomics checkout page form the invoice form

### PR DESCRIPTION
**Pull request by aadyanthaya**
Image of the invoice with the link to the Blockonomics  checkout page:
image

Image of the checkout page:
image
image

### CODE REVIEW

File: blockonomics-woocommerce.php [hunk 1]
```diff
@@ -59,9 +59,49 @@ function blockonomics_woocommerce_init()
...
+        $table_name = $wpdb->prefix . 'blockonomics_payments';
+        $query = $wpdb->prepare("SELECT expected_fiat,paid_fiat,currency FROM " . $table_name . " WHERE order_id = %d ", $order_id);
+        $results = $wpdb->get_results($query, ARRAY_A);
+        $paid_fiat = $blockonomics->calculate_total_paid_fiat($results);
+
+        if (!$paid_fiat) {
+            return $pay_url;
+        }
+        
+        if ($order->needs_payment()) {
```
Comment (DarrenWestwood) [inline blockonomics-woocommerce.php:98]: Let's first perform the checks on the woocommerce order object.

1. Perform checks on the Woocommerce Order object
2. Perform checks from Blockonomics settings/database

eg.
```
$payment_method = $order->get_payment_method();

if ($payment_method !== 'blockonomics' || !$order->needs_payment()) {
    return $pay_url;
}

$blockonomics = new Blockonomics();

if (!$blockonomics->is_partial_payments_active()) {
    return $pay_url;
}

$order_id = $order->get_id();
$paid_fiat = $blockonomics->get_order_paid_fiat($order_id);

if (!$paid_fiat) {
    return $pay_url;
}

return esc_url($blockonomics->get_order_checkout_url($order_id));
```

**Review by DarrenWestwood — COMMENTED** [review]
The changes are functioning as required for both the email and website flows. Links to pay underpaid orders lead directly to the Blockonomics select/payment page.

The display on some themes can be improved:
Current
Checkout UI Current

Desired
Checkout UI Fix

File: blockonomics-woocommerce.php (hunk 1 above)
Comment (aadyanthaya) [reply blockonomics-woocommerce.php:98]: Oh this is so good! Thanks for helping me refactor this.

File: blockonomics-woocommerce.php [hunk 2]
```diff
@@ -59,9 +59,49 @@ function blockonomics_woocommerce_init()
...
+        $is_partial_payments_active = $blockonomics->is_partial_payments_active();
+
+        if (!$is_partial_payments_active) {
+            return $pay_url;
+        }
+
+        global $wpdb;
+        $order_id = $order->get_id();
+        $table_name = $wpdb->prefix . 'blockonomics_payments';
+        $query = $wpdb->prepare("SELECT expected_fiat,paid_fiat,currency FROM " . $table_name . " WHERE order_id = %d ", $order_id);
```
[the remark this answers is not in the document]
Comment (aadyanthaya) [reply blockonomics-woocommerce.php]: don e with changes

File: blockonomics-woocommerce.php (hunk 1 above)
Comment (aadyanthaya) [reply blockonomics-woocommerce.php]: done with changes

**Review by DarrenWestwood — APPROVED** [review]

**Pull request merged**